src/platform/plugins/shared/navigation/public/top_nav_menu/top_nav_menu_item.tsx

```diff
@@ -68,29 +69,26 @@ export function TopNavMenuItem(props: TopNavMenuItemProps) {
     }
   }
 
-  const isModifiedEvent = (event: MouseEvent) =>
-    !!(event.metaKey || event.altKey || event.ctrlKey || event.shiftKey);
-
   function handleClick(event: MouseEvent<HTMLButtonElement | HTMLAnchorElement>) {
     if (isDisabled()) return;
-    if (props.href && isModifiedEvent(event)) return;
 
     props.run(event.currentTarget);
     if (props.isMobileMenu) {
       props.closePopover();
     }
   }
 
+  const routerLinkProps = getRouterLinkProps({ href: props.href, onClick: handleClick });
```

Comment:
```suggestion
  const routerLinkProps = props.href ? getRouterLinkProps({ href: props.href, onClick: handleClick }) : { onClick: handleClick };
```
I think we need to make this conditional on `props.href` to only use the guarded click handler when the item is a link, similar to what `handleClick` was doing before.

Reply:
Thanks for the catch! Pushing changes